test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/lang/ObjectJSTest.scala

```diff
@@ -41,4 +43,75 @@ class ObjectJSTest {
     val obj = new CloneOnNonScalaObject()
     assertThrows(classOf[CloneNotSupportedException], obj.boom())
   }
+
+  @Test def hashCodeOfSymbols(): Unit = {
+    /* None of the specific values here are by-spec. This test is highly
+     * implementation-dependent. It is written like this to make sure that we
+     * are returning different values for different arguments, but the specific
+     * values are irrelevant and could be changed at any time.
+     *
+     * By-spec, however, hashCode() delegates to System.identityHashCode() for
+     * symbols, since they are not Scala objects nor primitives that correspond
+     * to a hijacked class. So the values here must be equal to those in
+     * `SystemJSTest.identityHashCodeOfSymbols()`.
+     */
+
+    assumeTrue("requires JS symbols", Platform.jsSymbols)
+
+    @noinline def test(hash: Int, x: js.Symbol): Unit =
+      assertEquals(hash, x.hashCode())
+
+    // unfortunately, all symbols without description hash to the same value
+    test(0, js.Symbol())
+
+    test(0, js.Symbol(""))
+    test(-1268878963, js.Symbol("foobar"))
+    test(93492084, js.Symbol("babar"))
+    test(3392903, js.Symbol(null))
+
+    test(-1268878963, js.Symbol.forKey("foobar"))
+    test(93492084, js.Symbol.forKey("babar"))
+    test(3392903, js.Symbol.forKey(null))
+  }
+
+  @Test def hashCodeOfBigInts(): Unit = {
+    /* None of the specific values here are by-spec. This test is highly
+     * implementation-dependent. It is written like this to make sure that we
+     * are returning different values for different arguments, but the specific
+     * values are irrelevant and could be changed at any time.
+     *
+     * By-spec, however, hashCode() delegates to System.identityHashCode() for
+     * bigints, since they are not Scala objects nor primitives that correspond
+     * to a hijacked class (except for those that fit in a Long when we
+     * implement Longs as bigints). So the values here must be equal to those
+     * in `SystemJSTest.identityHashCodeOfBigInts()`.
+     *
+     * When we do implement Longs as bigints, the hashCode() of bigints that
+     * fit in a Long must conform to j.l.Long.hashCode(). That is however
+     * indepently tested in `LongTest.hashCodeTest()`.
```

Comment:
Do you think it would be worth having a test testing this more explicitly? (IIUC, right now this is only tested implicitly by having the same values in the `test` calls, easy to forget that they need to stay in sync IMO).

Reply:
They don't *have* to stay in sync. They will *have no choice* but to stay in sync, because the underlying implementation is the same (out of necessity). It's not possible to write an implementation that would behave differently here and in `LongTest.hashCodeTest()`.

Reply:
I see. Then I find this comment confusing. Maybe just remove it?

Also there's a typo: "indepently"

Reply:
> Then I find this comment confusing. Maybe just remove it?

Yes, indeed. I removed it.